package.json

```diff
@@ -56,9 +56,10 @@
     "broccoli-string-replace": "^0.1.2",
     "chalk": "^4.0.0",
     "debug": "^4.1.1",
+    "electron-devtools-installer": "^3.0.0",
```

Comment:
I wouldn't think we'd need these in `ember-electron`'s dependencies, because that would install it in the host Ember app, but we're already installing it in the Electron app using the blueprint.

Reply:
Without this here, the tests would go 💥. I agree with you, but wasn't sure what our alternatives were.

Reply:
Any ideas on how to make these not need to be dependencies, but still have the tests work?

Reply:
Not offhand...how do the tests fail without it?

Reply:
Cannot find module electron-devtools-installer, etc. The dependencies have to be installed for testing, it seems.

Reply:
Oh...right...because the blueprint generates `electron-app/tests/index.js` that does `require('ember-electron/lib/test-support/test-index')` and then the code running there will have a module search that is inside `ember-electron`, i.e. in the Ember app, not the Electron app.

I guess that's also why we need to have `electron-protocol-serve` as a dependency of `ember-electron`...ugh. That sucks because it means that the tests potentially run against a different version of things than the app (maybe not a huge deal for the devtools installer, but definitely not good for `electron-protocol-serve` and any other dependencies we might add in the future).

I guess the options I see are:
1. Move the code from that `test-index.js` module that lives in `ember-electron` into the blueprint-generated `tests/index.js` file.
2. Use `require.resolve()` to find the version of those dependencies that is in the electron app, using our knowledge of where it is that we get from `process.argv`
3. Leave it as-is and file an issue to address separately

I think I'd opt for a mix of (1) and (3), which would mean deleting the `installExtensions()` from `test-index.js` and adding it to `forge/files/tests/index.js` before the `require()`. Then at least we're not making the problem worse, and we can file an issue to sort out `electron-protocol-serve`, which will either result in doing something similar, or, I think more likely, getting rid of it entirely so we can have source maps post Electron 6.

Reply:
@bendemboski would it be okay with you if we handled this in a separate PR or did you want to do it in this one?

Reply:
I think/hope it would just be a quick cut-and-paste job to move the code into the `forge/files/tests/index.js` and then remove these dependencies -- can we try that, and if it doesn't work, then revert and file an issue?

In either case, I really don't think we need `electron-is-dev` here since we're not running that check in the test version of the extensions installation, so I think either way we should pull that from the `ember-electron` deps.

Reply:
@bendemboski I don't understand what to copy over. Don't we need the app so we can do `app.on('ready'`? Does that mean I should require electron in the other file? I'm happy to do whatever you think is best, but I do not know what to do exactly.

Reply:
Oh, okay, gotcha! Yeah, the test setup is a bit of a Rube Goldberg machine right now!

I'm saying to remove the code you added to `lib/test-support/test-index.js` and then make `forge/files/tests/index.js` look like

```javascript
const { app } = require('electron');
const { default: installExtension, EMBER_INSPECTOR } = require('electron-devtools-installer');

app.on('ready', function onReady() {
  installExtension(EMBER_INSPECTOR)
    .catch((err) => console.log('An error occurred: ', err));
});

require('ember-electron/lib/test-support/test-index');
```

I'm pretty sure that will allow us to remove the extra deps from `ember-electron`'s `package.json`.

### Explanation

When we run `ember electron:test` we build out the Ember app and instead of launching Electron and pointing to `/electron-app/src/index.js` as the main process code, we point to `/electron-app/tests/index.js` (both of these files are generated when the blueprint runs).

[forge/files/tests/index.js](https://github.com/adopted-ember-addons/ember-electron/blob/master/forge/files/tests/index.js) (which the blueprint uses to generate `/electron-app/tests/index.js`) is just one line:

```javascript
require('ember-electron/lib/test-support/test-index');
```

so it just executes exactly the code that lives in `/node_modules/ember-electron/lib/test-support/test-index.js`. But when that file tries to require modules, it's looking in `/node_modules` rather than `electron-app/node_modules`, so the dependencies need to be installed in the root Ember app rather than in the Electron app (`electron-app`).

So by moving the code to the blueprint-generated `electron-app/tests/index.js`, we allow it to require modules from `electron-app/node_modules`, i.e. the same set of modules that the non-test main process code has access to.

Reply:
Thanks for the explanation! I think I updated everything accordingly.